pkg/operator2/helpers.go

```diff
@@ -0,0 +1,21 @@
+package operator2
+
+import configv1 "github.com/openshift/api/config/v1"
+
+func moveLocalSecretToFileStringSource(ref configv1.LocalSecretReference) configv1.StringSource {
+	return configv1.StringSource{
+		StringSourceSpec: configv1.StringSourceSpec{
+			File: getLocalSecretFilename(ref),
+		},
+	}
+}
+
+// TODO: the logic of naming the CMs and Secrets to be mounted to the OSIN container
```

Comment:
Since we're using local references only, we don't have to worry about naming collisions between any of the secrets or configmaps referenced.
It's possible there's an existing secret or configmap in the target namespace. I don't see an easy way to guarantee there won't be any collisions; the ones I can think of would make syncing updates and deletions more annoying and difficult. Some OK solutions would be like appending the name with some moniker, `operator-config/mycm` -> `cluster-integrated-auth/mycm-clusterauth-v4` or blacklisting certain names.

It might be good to go the content-addressable route. Use a consistent hash of the data in the name + use ownerrefs to GC unused cm & secrets. @enj thoughts?

Reply:
I will explore options.